content/hardware/04.pro/boards/portenta-h7/tutorials/secure-boot/secure-boot.md

```diff
@@ -0,0 +1,71 @@
+
+### 1. Generate Custom Cecurity Keys
+The default keys provided with the mbed platform are obviously only intended for development purposes. In a production environment it is strongly recommended to generate a new key pair (public and private key).
+This can be done with [**imgtool**](https://github.com/arduino/imgtool-packing/releases/latest). You can download and install it directly from the release section.
+
+***`imgtool` is already installed by the mbed platform and can be found in the `%LOCALAPPDATA%\Arduino15\packages\arduino\tools\imgtool` directory on Windows, in `~/.arduino15/packages/arduino/tools/imgtool` on Linux and in `~/Library/Arduino15/packages/arduino/tools/imgtool` on macOS.***
+
+To generate the new keys you can use this command line:
+```
+imgtool keygen --key my-sign-keyfile.pem -t ecdsa-p256
+imgtool keygen --key my-encrypt-keyfile.pem -t ecdsa-p256
+```
+This command line will generate two private PEM encoded security keys and save them in the current directory with `my-sign-keyfile.pem` and `my-encrypt-keyfile.pem` names. The algorithm used to generate the keys is ECDSA 256bit.
+
+Remember to **save the keys on a secure location** and not to lose them.
+
+### 2. Upload the Custom Keys to the Board
+Once the keys have been generated they have to be uploaded on the Portenta. This procedure has to be done only once, because it's persistent. To extract the public\private key and encode it in to a "C" byte array inside a `.h` header file you can use:
+```
+imgtool getpriv -k my-encrypt-keyfile.pem > ecsda-p256-encrypt-key.h 
+imgtool getpub -k my-sign-keyfile.pem > ecsda-p256-signing-key.h
+```
+
+Now you have to replace the keys inside the Sketch to update the bootloader(**STM32H747_updateBootloader**).
+To do so just save the sketch to another location and replace the `ecsda-p256-encrypt-key.h` and `ecsda-p256-signing-key.h` files with the newly generated ones and then [update the bootloader](https://docs.arduino.cc/tutorials/portenta-h7/updating-the-bootloader) again.
+
+***NOTE: In case the keys are compromised, this process can be performed again with a new set of keys, but any firmware signed with the previous pair will no longer work.***
+
+### 3. Use the Custom Keys when Compiling
+Since the default keys have been changed in favour of custom generated ones, the new ones have to be used when compiling and uploading a sketch, because the compiled sketch is signed and encrypted using such keys.
+
+To override the security keys used during the compile you have to use the Arduino CLI and specify the keys with:
+```
+arduino-cli compile -b arduino:mbed_portenta:envie_m7 --board-options security=sien --keys-keychain <path-to-your-keys> --sign-key ecsdsa-p256-signing-key.pem --encrypt-key ecsdsa-p256-encrypt-key.pem /home/user/Arduino/MySketch
+```
+
+## Learn More
+If you want to implement secure boot for your platform [this](https://arduino.github.io/arduino-cli/latest/guides/secure-boot/) should be helpful.
```

Comment:
```suggestion
If you want to implement secure boot for your platform have a look at [this article] (https://arduino.github.io/arduino-cli/latest/guides/secure-boot/).
```

Reply:
You platform meaning...? Your board? Your core?

Reply:
https://arduino.github.io/arduino-cli/0.21/platform-specification/